File: app/src/main/java/com/example/dagger2demo/ExampleApp.java
package com.example.dagger2demo;

import android.app.Application;

import com.example.dagger2demo.dagger.AppComponent;
import com.example.dagger2demo.dagger.DaggerAppComponent;

public class ExampleApp extends Application {
   AppComponent appComponent;
    @Override
    public void onCreate() {
        super.onCreate();
      appComponent = DaggerAppComponent.create();
    }
    public AppComponent getActivityComponent(){
        return appComponent;
    }
}

File: app/src/main/java/com/example/dagger2demo/ExampleApp.kt
package com.example.dagger2demo

import android.app.Application
import com.example.dagger2demo.dagger.AppComponent
import com.example.dagger2demo.dagger.DaggerAppComponent

class ExampleApp : Application() {
    var appComponent: AppComponent? = null
    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerAppComponent.create()
    }
}